pkg/sql/schemachanger/scpb/elements.proto

```diff
@@ -529,3 +530,65 @@ message IndexData {
 message TablePartitioning {
   uint32 table_id = 1 [(gogoproto.customname) = "TableID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
 }
+
+message Function {
+  message Parameter {
+    option (gogoproto.equal) = true;
+    cockroach.sql.catalog.catpb.FunctionParamClass class = 1 [(gogoproto.nullable) = false];
+    string name = 2;
+    sql.sem.types.T type = 3;
+    string default_expr = 4;
+  }
+
+  message ReturnType {
+    option (gogoproto.equal) = true;
+    sql.sem.types.T type = 1;
+    bool return_set = 2;
+  }
+  uint32 function_id = 1 [(gogoproto.customname) = "FunctionID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+  repeated Parameter params = 2 [(gogoproto.nullable) = false];
+  ReturnType return_type = 3 [(gogoproto.nullable) = false];
+}
+
+message FunctionName {
+  uint32 function_id = 1 [(gogoproto.customname) = "FunctionID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+  string name = 2;
+}
+
+message FunctionBody {
+  uint32 function_id = 1 [(gogoproto.customname) = "FunctionID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+  string function_body = 2;
```

Comment:
Similarly to what `Expression` and `TypeT` do, you should explicitly list all the references in the function body. I expect there to be type descriptor IDs, sequence descriptor IDs, function descriptor IDs, table/view + column ID pairs, at least.